### PR TITLE
Updating Communication Services SDK version

### PR DESCRIPTION
**Pull request by matthohn-msft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/CommunicationServiceResource.cs [hunk 1]
```diff
@@ -272,23 +268,19 @@ public virtual async Task<ArmOperation<CommunicationServiceResource>> UpdateAsyn
         /// </item>
         /// </list>
         /// </summary>
-        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
         /// <param name="patch"> Parameters for the update operation. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="patch"/> is null. </exception>
-        public virtual ArmOperation<CommunicationServiceResource> Update(WaitUntil waitUntil, CommunicationServiceResourcePatch patch, CancellationToken cancellationToken = default)
```
Comment (archerzz) [inline sdk/communication/Azure.ResourceManager.Communication/src/Generated/CommunicationServiceResource.cs:279]: Breaking change. Even in `1.0.0`, the update operation is a long running operation, right?

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/CommunicationServiceResource.cs [hunk 2]
```diff
@@ -432,23 +424,19 @@ public virtual Response<CommunicationServiceKeys> GetKeys(CancellationToken canc
         /// </item>
         /// </list>
         /// </summary>
-        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
         /// <param name="content"> Parameter that describes the Regenerate Key Operation. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
-        public virtual async Task<ArmOperation<CommunicationServiceKeys>> RegenerateKeyAsync(WaitUntil waitUntil, RegenerateCommunicationServiceKeyContent content, CancellationToken cancellationToken = default)
```
Comment (archerzz) [inline sdk/communication/Azure.ResourceManager.Communication/src/Generated/CommunicationServiceResource.cs:439]: Breaking change. Is this just a breaking change to previous beta versions?

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/CommunicationServiceResourceCollection.cs [hunk 3]
```diff
@@ -37,8 +39,14 @@ protected CommunicationServiceResourceCollection()
         /// <summary> Initializes a new instance of the <see cref="CommunicationServiceResourceCollection"/> class. </summary>
         /// <param name="client"> The client parameters to use in these operations. </param>
         /// <param name="id"> The identifier of the parent resource that is the target of operations. </param>
-        internal CommunicationServiceResourceCollection(ArmClient client, ResourceIdentifier id) : base(client, id)
```
Comment (archerzz) [inline sdk/communication/Azure.ResourceManager.Communication/src/Generated/CommunicationServiceResourceCollection.cs:40]: I think this is unexpected changes due to the type of `subscriptionid` was changed from `string` to `Guid`.

**Review by archerzz — COMMENTED** [review]
There are two types of issues here:
- Looks like `Azure.ResourceManager.Communication` has an unexpected release of `1.0.0` which doesn't follow track2 mgmt SDK guidelines. So the `1.1.0` will have a lot breaking changes anyway.
- Due to upgrade to V5 common types, now `SubscriptionId` has been changed from `string` type to `Guid` type, which triggers cascading changes. We need to decide how to deal with V5 upgrade.

@ArthurMa1978 @m-nash FYI

**Comment by archerzz** [comment]
I've filed an issue regarding the v5 common type regression: https://github.com/Azure/autorest.csharp/issues/3210

File: sdk/communication/Azure.Communication.JobRouter/src/Models/ClassificationPolicy.cs [hunk 4]
```diff
@@ -9,7 +9,10 @@
 namespace Azure.Communication.JobRouter.Models
 {
     [CodeGenModel("ClassificationPolicy")]
-    [CodeGenSuppress("ClassificationPolicy")]
+    [
+#pragma warning disable CA1825 // Avoid zero-length array allocations
```
[the remark this answers is not in the document]
Comment (archerzz) [reply sdk/communication/Azure.Communication.JobRouter/src/Models/ClassificationPolicy.cs:13]: Probably don't need this change after this PR is merged: https://github.com/Azure/azure-sdk-for-net/pull/34780

File: sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/ProvisioningState.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Communication.Models
+{
+    /// <summary> Provisioning state of the resource. Unknown is the default state for Communication Services. </summary>
+    public readonly partial struct ProvisioningState : IEquatable<ProvisioningState>
```
Comment (archerzz) [inline sdk/communication/Azure.ResourceManager.Communication/src/Generated/Models/ProvisioningState.cs:14]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/communication/.editorconfig [hunk 6]
```diff
@@ -0,0 +1,2 @@
+[*.{cs,vb}]
+dotnet_diagnostic.CA1825.severity = none
```
Comment (m-nash) [inline sdk/communication/.editorconfig:2]: any reason we are suppressing this instead of fixing it?

**Comment by m-nash** [comment]
> There are two types of issues here:
> 
> * Looks like `Azure.ResourceManager.Communication` has an unexpected release of `1.0.0` which doesn't follow track2 mgmt SDK guidelines. So the `1.1.0` will have a lot breaking changes anyway.
> * Due to upgrade to V5 common types, now `SubscriptionId` has been changed from `string` type to `Guid` type, which triggers cascading changes. We need to decide how to deal with V5 upgrade.
> 
> @ArthurMa1978 @m-nash FYI

Yes 1.0.0 was released before it was ready.  At the time we discussed the best path forward and it was determined that we would start with 1.1.0-beta* and when this GA's it will introduce many breaking changes compared to 1.0.0.

File: sdk/communication/.editorconfig (hunk 6 above)
Comment (matthohn-msft) [reply sdk/communication/.editorconfig]: I have reverted this file.  It was for my local env only, this should not be a problem in the pipeline.

**Review by archerzz — APPROVED** [review]
LGTM

**Pull request merged**